bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceDialog.java

```diff
@@ -1065,7 +1065,7 @@ private void updateButtonState(boolean disableReplace) {
 			boolean isSelectionGoodForReplace = selectionString != "" //$NON-NLS-1$
 					|| !isRegExSearchAvailableAndActive;
 
-			fWholeWordCheckBox.setEnabled(isWord(str) && !isRegExSearchAvailableAndActive);
+			fWholeWordCheckBox.setEnabled(findReplaceLogic.isWholeWordSearchAvailable(getFindString()));
```

Comment:
I would expect all places setting the enablement of the whole-word checkbox to be adapted accordingly. In particular, I would expect line 1181 to be adapted as well:
```java
fWholeWordCheckBox.setEnabled(!findReplaceLogic.isRegExSearchAvailableAndActive());
```